pkg/server/event_handler_test.go

```diff
@@ -325,18 +370,40 @@ func newTestServer(handler func(w http.ResponseWriter, r *http.Request)) (*http.
 	}
 }
 
+// Returns fake metadata that include github resource info.
+func getFakeMetadata() []byte {
+	return []byte(`{
+		"metadata": {
+		  "git-commit": "test-github-commit",
+		  "triggered-timestamp": "2023-04-25T17:44:57",
+		  "git-workflow-sha": "test-workflow-sha",
+		  "git-workflow": "test-workflow",
+		  "git-repo": "test-github-repo"
+		}
+	  }`)
+}
+
 // Returns a fake http func that writes the data in http response.
-func handleObjectRead(t *testing.T, data []byte) func(w http.ResponseWriter, r *http.Request) {
+func handleObjectRead(t *testing.T, data, metadata []byte) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I am not sure why we need this func to handle object metadata, aren't the object metadata in notification payload? This func should only handles GCS return a object.